internal/pkg/service/common/etcdop/consumer_test.go

```diff
@@ -0,0 +1,130 @@
+package etcdop
+
+import (
+	"context"
+	"net"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/keboola/go-utils/pkg/wildcards"
+	gonanoid "github.com/matoous/go-nanoid/v2"
+	"github.com/stretchr/testify/assert"
+	"google.golang.org/grpc"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/log"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/etcdhelper"
+)
+
+// nolint:paralleltest // the test run the "compact" operation and breaks the other tests running in parallel
+func TestWatchConsumer(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), 15*time.Second)
+	defer cancel()
+	wg := &sync.WaitGroup{}
+	logger := log.NewDebugLogger()
+
+	// Both clients must use same namespace
+	etcdNamespace := "unit-" + t.Name() + "-" + gonanoid.Must(8)
+
+	// Create client for the test
+	testClient := etcdhelper.ClientForTestWithNamespace(t, etcdNamespace)
+
+	// Create watcher client with custom dialer
+	var conn net.Conn
+	dialerLock := &sync.Mutex{}
+	dialer := func(ctx context.Context, s string) (net.Conn, error) {
+		dialerLock.Lock()
+		defer dialerLock.Unlock()
+		var err error
+		conn, err = (&net.Dialer{}).DialContext(ctx, "tcp", s)
+		return conn, err
+	}
+	c := etcdhelper.ClientForTestWithNamespace(t, etcdNamespace, grpc.WithContextDialer(dialer))
+
+	// Create consumer
+	pfx := prefixForTest()
+	init := pfx.
+		GetAllAndWatch(ctx, c).
+		SetupConsumer(logger).
+		WithOnCreated(func(header *Header) {
+			logger.Infof(`OnCreated: created (rev %v)`, header.Revision)
+		}).
+		WithOnRestarted(func(reason string, delay time.Duration) {
+			logger.Infof(`OnRestarted: %s`, reason)
+		}).
+		WithForEach(func(events []WatchEvent, header *Header, restart bool) {
+			var str strings.Builder
+			for _, e := range events {
+				str.WriteString(e.Type.String())
+				str.WriteString(` "`)
+				str.Write(e.Kv.Key)
+				str.WriteString(`", `)
+			}
+			logger.Infof(`ForEach: restart=%t, events(%d): %s`, restart, len(events), strings.TrimSuffix(str.String(), ", "))
+		}).
+		StartConsumer(wg)
```

Comment:
Takto sa pouziva `WatchConsumer`, namiesto toho dlheho `for+switch` co bol vyssie.
`WithForEach` sa pouzije asi vzdy,  ... `WithOnCreated/WithOnRestarted` ak treba.